docgen/src/guide/Refining_hits.md

```diff
@@ -0,0 +1,551 @@
+---
+title: Refining within hits
+mainTitle: Guides
+layout: main.pug
+category: guide
+navWeight: # idk what to put here 🚨
+---
+
+## Introduction
+
+In some interfaces, you want to have buttons _inside_ your Hits that will change the refinement of the current search, rather than as standalone components. Since we can't replicate this pattern with widgets, we will need to go with Refinementlists that don't render anything. Please note that this is an advanced tutorial and requires moderate knowledge of how React and React InstantSearch work. Our final result will look something like this, let's get started!
+
+<iframe src="https://codesandbox.io/embed/oY1klpZYB" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+## Regular setup
+
+First, we'll set up a regular project wit create-react-app and react-instantsearch.
+
+```sh
+$ yarn create react-app refining-hits # or npm install -g create-react-app && create-react-app refining-hits
+$ cd refining-hits
+$ yarn add react-instantsearch # or npm install --save react-instantsearch
+```
+
+The index that we will use in this example needs to be set up with: 
+
+```jsx
+index.setSettings({
+  attributesForFaceting: ['searchable(keywords)', 'searchable(owner.name)'],
+});
+```
+
+An index that already has these settings is the index for search on [Yarn](https://yarn.fyi), which is used in this guide. We will get started in `App.js` and set up a regular React InstantSearch setup:
```

Comment:
In this guide we will use the search index we use on [Yarn](https://yarn.fyi). It already has the mandatory `attributesForFaceting ` configured.
We will get started in `App.js` and set up a regular React InstantSearch app: